ui/model/action-result.ts

```diff
@@ -0,0 +1,29 @@
+export interface ActionResult<T> {
```

Comment:
That's interesting. So that's some kind of template. Even though it's called an interface. Interesting. Seems like a good idea to use this for api results.

Reply:
It is a generic interface. It made sense since we had a lot of common structure and it didn't make any sense creating multiple classes to model them. Instead we can only model the payload :)